src/Symfony/XmlParameterMapFactory.php

```diff
@@ -40,18 +42,23 @@ public function create(): ParameterMap
 
 		/** @var Parameter[] $parameters */
 		$parameters = [];
-		foreach ($xml->parameters->parameter as $def) {
-			/** @var SimpleXMLElement $attrs */
-			$attrs = $def->attributes();
 
-			$parameter = new Parameter(
```

Comment:
What are these changes for? Can you remove them?

Reply:
This is a fix for handling XML files without `<services>` / `<parameters>`, I stumbled upon this when tests read container from files and I did not want to declare both services and parameters in each test case. I believe it's not needed now for _my_ usage, but it still fixes potential problem with using undefined properties. Let me know if you want to keep it or not.

Reply:
@ondrejmirtes for a wider context: if I understand [Symfony's container schema](https://symfony.com/schema/dic/services/services-1.0.xsd) both `parameters` and `services` nodes may occur 0 or 1 times (`minOccurs="0"` without `maxOccurs`), so it's not only a theoretical fix, but a real one. Of course, in practice it's not much likely that container does not have services and/or parameters, but technically speaking it's possible 🙂.